### ISSUE TITLE
Suppress "Continue Reading" from blog entries on index pages if they do not have an excerpt marker

### ISSUE THREAD

**Issue by wez** [squidfunk/mkdocs-material#6783]
### Context

See the discussion in https://github.com/squidfunk/mkdocs-material/discussions/6759

### Description

If I use the  marker to define the cut point for an excerpt, the index page works as I'd expect; the content before the marker is used as an excerpt and then the "Continue Reading" link takes the visitor to the full content for that post. All good.

However, if I omit the  marker then the entire content is used in the index page (which is what I'd expect), AND "Continue Reading" is shown below it. That gives the visitor the expectation that there is more to read in that case, but when they click through they are disappointed.

I propose that the "Continue Reading" link be omitted in that second case.

### Related links

* https://github.com/squidfunk/mkdocs-material/discussions/6759

### Use Cases

See discussion in https://github.com/squidfunk/mkdocs-material/discussions/6759

### Before submitting

- [X] I have read and followed the [change request guidelines](https://squidfunk.github.io/mkdocs-material/contributing/requesting-a-change/).
- [X] I have verified that [my idea is a change request and not a bug report](https://squidfunk.github.io/mkdocs-material/contributing/requesting-a-change/#its-not-a-bug-its-a-feature).
- [X] I have ensured that, to the best of my knowledge, [my idea will benefit the entire community](https://squidfunk.github.io/mkdocs-material/contributing/requesting-a-change/#benefit-for-the-community).
- [X] I have included relevant links to [the documentation](https://squidfunk.github.io/mkdocs-material/), related [issues](https://github.com/squidfunk/mkdocs-material/issues), and [discussions](https://github.com/squidfunk/mkdocs-material/discussions) to underline the need for my idea.

**Comment by squidfunk** [comment]
Fixed in b7fc7bad5. The `post.more` attribute now contains the content after the post separator. Template authors can decide to render the entire content on the overview page and make it expandable. This allows for more flexibility than just providing a boolean indicator, which is why we went for it. The continue reading link should now be automatically removed if there is no separator and/or nothing more to read (= the separator is at the very end of the Markdown).
